src/webui/api/torrentscontroller.cpp

```diff
@@ -252,6 +254,25 @@ namespace
             idList << BitTorrent::TorrentID::fromString(hash);
         return idList;
     }
+
+    nonstd::expected<QUrl, QString> validateWebseedUrl(const QString &urlStr)
+    {
+        const QString normalizedUrlStr = QUrl::fromPercentEncoding(urlStr.toUtf8());
```

Comment:
I believe `toLatin1()` should be used instead of `toUtf8()`. Isn't encoded URL supposed to contain only ASCII characters?

Reply:
https://stackoverflow.com/questions/912811/what-is-the-proper-way-to-url-encode-unicode-characters

Reply:
Will switch to `toLatin1()` ([context](https://github.com/qbittorrent/qBittorrent/pull/21056#discussion_r1705015747)).